scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/DownloadJobXMLServlet.java

```diff
@@ -66,31 +66,30 @@ private void getJobXML(HttpServletRequest request, HttpServletResponse response)
         String jobId = request.getParameter("jobId");
         String sessionId = request.getParameter("sessionId");
         String fileName = "job_" + jobId + ".xml";
-        String location = "job" + jobId + ".result";
 
-        ServletOutputStream out = null;
-        try (InputStream is = new ByteArrayInputStream(((SchedulerServiceImpl) Service.get()).getJobXML(sessionId,
-                                                                                                        jobId)
-                                                                                             .getBytes(StandardCharsets.UTF_8))) {
+        try (ServletOutputStream out = response.getOutputStream();
+                InputStream is = new ByteArrayInputStream(((SchedulerServiceImpl) Service.get()).getJobXML(sessionId,
+                                                                                                           jobId)
+                                                                                                .getBytes(StandardCharsets.UTF_8))) {
             response.setContentType("application/xml");
             response.setHeader("Content-disposition", "attachment; filename=" + fileName);
-            response.setHeader("Location", location);
-
-            out = response.getOutputStream();
+            response.setHeader("Location", fileName);
 
             int buffer;
             while ((buffer = is.read()) != -1) {
                 out.write(buffer);
+                out.flush();
             }
             LOGGER.debug("Successfully downloaded job XML for job: {}", jobId);
 
-        } catch (Exception t) {
-            LOGGER.warn("Failed to download workflow", t);
-            String str = "Failed to download workflow: " + JSONUtils.getJsonErrorMessage(t);
-            Objects.requireNonNull(out).write(str.getBytes());
-        } finally {
-            Objects.requireNonNull(out).flush();
-            out.close();
+        } catch (RestServerException e) {
```

Comment:
Are we sure that a RestServerException is always about permissions?

Reply:
@jrochas Gleb and I conveyed that the server API will always return a 403 error in case of permission issues, but here I don't test the if the http status code is 403, I just display it.
You're right, I forgot to do what I do in this catch only if it's 403.
Thanks justine